packages/components/src/components/TokenInput/TokenInput.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  BoxProps,
+  FormControl,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputProps,
+  InputRightElement,
+  Select,
+  SelectFieldProps,
+} from '@chakra-ui/react';
+import React from 'react';
+import { defaultTokens } from './tokens';
+
+export interface Token {
+  // token sybmol
+  symbol: string;
+  // token contract address
+  address: string;
+}
+
+export interface TokenInputProps {
+  /**
+   * an array of tokens to display
+   */
+  tokens?: Token[];
+  /**
+   * value of the input
+   */
+  value: string;
+  /**
+   * onChange handler
+   */
+  onValueChange: (value: string) => void;
+  /**
+   * label for the input
+   */
+  label?: string;
+  /**
+   * selected token
+   */
+  selectedToken: string;
+  /**
+   * token change hanlder
+   */
+  onTokenChange: (token: string) => void;
+  /**
+   * want to show default tokens
+   */
+  showDefualtTokens?: boolean;
+  /**
+   * style props for the input
+   */
+  inputProps?: InputProps;
+  /**
+   * style props for the select
+   */
+  selectProps?: SelectFieldProps;
+}
+
+/**
+ * A number input component that is used to enter number of tokens and also select the token from a list of tokens.
+ */
+
+export const TokenInput: React.FC<TokenInputProps & BoxProps> = ({
+  tokens = [],
+  value,
+  onValueChange,
+  label,
+  selectedToken,
+  onTokenChange,
+  showDefualtTokens = true,
```

Comment:
If we keep this prop, we should probably default it to `false`. imo it'd be unexpected and confusing to render a `select` and have `options` appear that you didn't specify.